### PR TITLE
Part 4 done

### PR DESCRIPTION
**Pull request by shaefelinger**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -258,23 +258,21 @@ bool MobilePhone::updateOperatingSystem(float osVersion)
 
 struct TvScreen
 {
-    TvScreen();
+    TvScreen() {
+        std::cout << "CONSTRUCTING TvScreen" << std::endl;
+    }
```
Comment (matkatmusic) [inline main.cpp]: Curly braces go on their own line
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -145,6 +145,7 @@ struct Computer
 
 Computer::Computer()
 {
+    amountOfRam = 8;
```
Comment (matkatmusic) [inline main.cpp]: When you have to choose where to initialize your member variables and your choices are:
Constructor Member Initializer List
Inside the Constructor's body
Try to use the Constructor Member Initializer List.
The reason is that the member variables need to be instantiated/initialized when the class is created.
When you assign to them in the constructor body, you are potentially re-initializing them, which is a waste of resources.
Take a look at the output of this program: https://godbolt.org/z/q83YM63ze
Take note of the second instance being created and destroyed due to that constructor body assignment.
- [x] revise, use the member initializer list to initialize your member variables

**Review by matkatmusic — CHANGES_REQUESTED** [review]
In addition to the things mentioned below, you did not do instruction 2)
```
 2) make some of your member functions use those initialized member variables via std::cout statements.
```

File: main.cpp (hunk 1 above)
Comment (shaefelinger) [reply main.cpp]: oh, i guess i have written too much Javascript… ;-)

File: main.cpp [hunk 3]
```diff
@@ -251,14 +252,15 @@ void MobilePhone::charge()
 
 bool MobilePhone::updateOperatingSystem(float osVersion)
 {
-    std::cout << "Updated Operating System to Version " << osVersion << std::endl;
+    std::cout << "Updated Operating System to Version " << osVersion << " on this " << MobilePhone::manufacturer << "-phone from " << MobilePhone::yearOfManufacture << std::endl;
```
Comment (matkatmusic) [inline main.cpp]: Don't use the fully-qualified name when accessing member variables inside member functions. 
You should only use the fully-qualified name when using **static** class member variables.
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**